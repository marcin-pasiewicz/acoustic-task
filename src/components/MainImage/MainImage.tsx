import React, { ReactElement } from 'react'
import styled from 'styled-components'
import { CONTENT_BASE_URL } from '../../config/api'

interface Props {
  value: any;
}

export default function MainImage({ value }: Props): ReactElement {
  const { altText = '', fileName = '' } = value.leadImage.asset;
  const src = `${CONTENT_BASE_URL}${value.leadImage.url}`
  return (
    <StyledImage data-test-id="main-image" src={src} alt={altText || fileName} />
  )
}

const StyledImage = styled.img.attrs(props => ({
  ...props,
  alt: props.alt,
  src: props.src,
}))`
  width: 100%;
  height: auto;
`