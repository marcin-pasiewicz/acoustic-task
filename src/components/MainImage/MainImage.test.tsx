import React from 'react';
import { render } from '@testing-library/react';
import MainImage from './MainImage';

describe('<MainImager/>', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<MainImage
      value={{
        leadImage: {
          url: 'some url',
          asset: {
            altText: 'altText',
            fileName: 'fileName',
          }
        }}
      } />);
    expect(asFragment()).toMatchSnapshot();
  });
});
