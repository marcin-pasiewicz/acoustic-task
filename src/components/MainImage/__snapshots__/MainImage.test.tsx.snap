// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainImager/> renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="altText"
    class="sc-AxjAm bReqtd"
    src="https://content-eu-4.content-cms.com/some url"
  />
</DocumentFragment>
`;
